hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java

```diff
@@ -388,6 +388,41 @@ public void testHiveTableAndIcebergTableWithSameName(TableType tableType)
     HIVE_METASTORE_EXTENSION.metastoreClient().dropTable(DB_NAME, hiveTableName);
   }
 
+  @Test
+  public void testTableExists() throws TException, IOException {
+    String testTableName = "test_table_exists";
+    TableIdentifier identifier = TableIdentifier.of(DB_NAME, testTableName);
+    TableIdentifier metadataIdentifier = TableIdentifier.of(DB_NAME, testTableName, "partitions");
+
+    assertThat(catalog.tableExists(identifier))
+        .as("Table should not exist before create")
+        .isFalse();
+    catalog.buildTable(identifier, SCHEMA).create();
+    assertThat(catalog.tableExists(identifier)).as("Table should exist after create").isTrue();
+    assertThat(catalog.tableExists(metadataIdentifier))
+        .as("Metadata table should also exist")
+        .isTrue();
+
+    boolean dropped = catalog.dropTable(identifier);
```

Comment:
nit: instead of introducing a variable could you merge this line with the following similarly as you do at L397?

Reply:
generally I tried to avoid assert on statement which might have side effect such as drop a table, but I do observe the similar pattern in other unit tests so updated as suggested